client.js

```diff
@@ -242,6 +256,25 @@ WemoClient.prototype._subscribe = function(serviceType) {
       setTimeout(this._subscribe.bind(this), 120 * 1000, serviceType);
     }
   }.bind(this));
+  
+  req.on('error', function(err) {
+    // We can't pass back errors to the calling module so we'll do what we can to try
+    // and gracefully recover from HTTP errors.
+    // ECONNREFUSED suggests that the port number may have changed
+    // EHOSTUNREACH suggests the device has gone (switched off maybe)
+    // ETIMEDOUT    seems to be recoverable - just lost it for a bit, we'll retry.
+    var timeout = 5; //seconds before we retry
+    console.log("HTTP Error (%s) occurred (re)subscribing to Wemo Device (%s - %s:%s), retrying.", 
```

Comment:
This worried me also - problem with debug is that the user has no idea there has been an issue inn wemo-client. What is we passed in an optional pointer to a log function to be used in the wemo-client constructer?